worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -20,15 +20,63 @@ if(!ACTIONS){
 }
 
 var URL = Java.type("java.net.URL");
+var MDC = Java.type("org.slf4j.MDC");
+var UUID = Java.type("java.util.UUID");
 
-function onProcessTask() {
+function onProcessTask(e) {
+    addMdcLoggingData(e);
     thisScript.install();
 }
 
+function addMdcLoggingData(e) {
+    // The logging pattern we use uses a tenantId and a correlationId:
+    // 
+    // https://github.com/CAFapi/caf-logging/tree/4ef35ae3a6da4329a427667782f7aaff4fee8c1d#pattern
+    // https://github.com/CAFapi/caf-logging/blob/4ef35ae3a6da4329a427667782f7aaff4fee8c1d/src/main/resources/logback.xml#L27
+    //
+    // This function adds a tenantId and correlationID to the MDC (http://logback.qos.ch/manual/mdc.html), so that log messages 
+    // from workers in the workflow will contain these values.
+    //
+    // See also addMdcData in WorkflowWorker, which performs similar logic to ensure log messages from the workflow-worker itself also 
+    // contain these values. 
+    
+    // Get MDC data from custom data.
+    var tenantId = e.task.getCustomData("tenantId");
+    var correlationId = e.task.getCustomData("correlationId");
+    
+    // Generate a random correlationId if it doesn't yet exist.
+    if (!correlationId) {
+        correlationId = UUID.randomUUID().toString();  
+    }
+
+    // Only if this worker is NOT a bulk worker; add tenantId and correlationId to the MDC.
+    if (!isBulkWorker(e)) {  
+        MDC.put("tenantId", tenantId);
+        MDC.put("correlationId", correlationId);
+    }
+    
+    // Add MDC data to custom data so that its passed it onto the next worker.
+    e.task.getResponse().getCustomData().put("tenantId", tenantId);
```

Comment:
Pull responseCustomData out into a variable so that you don't have to get it twice.